### ISSUE TITLE
Determining settings for WIKI Updater

### ISSUE THREAD

**Issue by scottforsyth** [rfennell/AzurePipelines#618]
### Azure DevOps Extension you are using
- Git based WIKI Updater

### Where are you running it?
- Azure DevOps Service (VSTS)

### Version of Extension/Task
- 1.7.5

### Question

I'm receiving the following error message:
#### Error: Error: spawn git ENOENT

I haven't been able to determine the right settings yet. I believe that the two that I need help with are the Git Name and Git Email. I don't have a .gitattributes file already, so I'm not sure what to be looking for. Where do I get the Git Name and Git Email? I'm using Azure DevOps for the git repro storage.

The Wiki clone repository URL is:
https://dwna@dev.azure.com/dwna/TestProject/_git/TestProject.wiki

In the error logs, it shows the URL used is in this format:
https://sfo@myemail.com:***@dev.azure.com/dwna/TestProject/_git/TestProject.wiki

So it appears that it's piecing together the URL from the Authentication section's Username since I have a different email for the Git Email. Also, The :*** seems odd.

I've tried both with the build agent credentials and with my account.

Here are my current settings.

![image](https://user-images.githubusercontent.com/5747160/73655724-9b05b700-465c-11ea-938e-cbb4acb8dca5.png)

**Comment by rfennell** [comment]
I see you are trying to write to an Azure DevOps WIKI Git Repo. The credentials required are not obvious. I keep trying to improve the [documentation](https://github.com/rfennell/AzurePipelines/wiki/WIKI-Updater-Tasks) but I am not there yet. The summary is...

### Using pass credentials
- Username is the name of the Azure DevOps organisation, so if you are using https://dev.azure.com/richardfennell then the username is richardfennell
- The password is a PAT token for a user with suitable rights

![image](https://user-images.githubusercontent.com/324718/73754007-93790780-475b-11ea-9174-3aff15abec69.png)

### If using the agents credentials
Just tick the box in the task
![image](https://user-images.githubusercontent.com/324718/73754076-b4d9f380-475b-11ea-8e7b-d52000ee0199.png)

But remember that the Agent Phase properties must have the use Oauth token box checked too

![image](https://user-images.githubusercontent.com/324718/73754163-dcc95700-475b-11ea-8cca-312ce9e759b4.png)

Hope this helps, let me know

**Comment by scottforsyth** [comment]
Thanks for your reply. That gets me closer. I feel confident in my sessions for the credentials now and I understand the URL pattern.

However, I'm still getting the same error. That error appears to be because it can't locate Git. I believe that my issue is the Git name and Git Email. Where do I locate the Git name and Git Email?

**Comment by rfennell** [comment]
I believe that the Git Name and Git Email should not be important, they are stored in the Git config file equiv. to

```
$ git config --global user.email "email@example.com"
````

and are not used for authentication.

I have looked at the logs for my test harness and with these variables (using creds) 

```
##[debug]gitname=builduser
##[debug]gitemail=build@demo
##[debug]user=richardfennell         < org nam
##[debug]password=***                  < PAT
##[debug]useAgentToken=False
Variable: Repo [dev.azure.com/richardfennell/MyProject/_git/MyProject.wiki]
```

this is see (where **** is a valid users PAT)
```
 URL used https://richardfennell:***@dev.azure.com/richardfennell/MyProject/_git/MyProject.wiki
```

If if using a Agent Token

```
##[debug]gitname=builduser
##[debug]gitemail=build@demo
##[debug]user=richardfennell               < not used
##[debug]password=null                       < not used
##[debug]useAgentToken=true
Variable: Repo [dev.azure.com/richardfennell/MyProject/_git/MyProject.wiki]
```
this is seen, where 'buildagent' is just any text it is no important, and *** is the System Token

```
URL used https://buildagent:***@dev.azure.com/richardfennell/GitHub/_git/GitHub.wiki
```

Sorry I can do more than offer these logs to compare. 

You could try to switch authentication model

**Comment by scottforsyth** [comment]
Thanks again @rfennell. And thanks for your app. I'm looking forward to using it.

I finally figured out what I was doing wrong. I had created a _Deployment Group Job_ for another purpose, and added the Git based WIKI Updater task to that job. Once I re-read the docs I noticed the subtle difference, I created an _Agent Job_ instead, and moved the task under it. After that, it works perfectly. 

I also confirmed that the Git name and email fields aren't important. I entered "NotUsed" for those two fields and everything continues to work.
